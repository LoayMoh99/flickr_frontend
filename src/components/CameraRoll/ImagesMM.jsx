/* eslint-disable linebreak-style */
/* eslint-disable max-len */
/* eslint-disable linebreak-style */
/* eslint-disable linebreak-style */
/* eslint-disable no-undef */
/* eslint-disable linebreak-style */
/* eslint-disable react/no-this-in-sfc */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable linebreak-style */
/* eslint-disable max-len */
/* eslint-disable linebreak-style */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable linebreak-style */
/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
/* eslint-disable linebreak-style */
import React from 'react';
import './ImagesMM.css';

const ImagesMM = (props) => {
  const { url } = props;
  return (
    <>
      <div className="column_adj_modal">
        <img className="img-responsive" id="content_modal" src={url} alt="image_flickr" />
      </div>
    </>
  );
};
export default ImagesMM;
