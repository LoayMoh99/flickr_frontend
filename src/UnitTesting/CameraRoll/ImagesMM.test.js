import React from 'react';
import ReactDOM from 'react-dom';
import ImagesMM from '../../components/CameraRoll/ImagesMM';
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

it('ImagesMM renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<ImagesMM
        url="https://picsum.photos/600/200?random=2"
      />, div);


});

it('ImagesMM renders correctly enzyme', () => {
    const wrapper = shallow(<ImagesMM
        url="https://picsum.photos/600/200?random=2"
      />,)
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("ImagesMM Should have an image", ()=>{
    const wrapper = shallow(<ImagesMM
        url="https://picsum.photos/600/200?random=2"
      />,)
    const image= wrapper.find("img");
    expect(image).toHaveLength(1);
})