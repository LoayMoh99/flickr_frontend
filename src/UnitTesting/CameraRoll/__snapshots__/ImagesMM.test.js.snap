// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagesMM renders correctly enzyme 1`] = `
<Fragment>
  <div
    className="column_adj_modal"
  >
    <img
      alt="image_flickr"
      className="img-responsive"
      id="content_modal"
      src="https://picsum.photos/600/200?random=2"
    />
  </div>
</Fragment>
`;
